File: packages/javascript/lib/word/randMusic.ts
import type { RandThemedWordOptions, WordDetail } from '../_types/global.js';
import { themedWord } from './wordGenerator.js';

/**
 * Music: instruments, forms and the words around them.
 *
 * `randWord({ theme: 'music' })` with the theme already chosen; every other
 * option is the same.
 *
 * @example
 * randMusic({ language: 'ko', count: 3 }); // ['피아노', '거문고', '교향곡']
 * randMusic({ language: 'en', count: 3 }); // ['Piano', 'Fiddle', 'Symphony']
 */
export function randMusic(options?: RandThemedWordOptions & { output?: 'value' }): string[];
/**
 * Music words with the language and theme each one came from.
 *
 * @example
 * randMusic({ language: 'ko', output: 'detail' });
 * // [{ word: '피아노', language: 'ko', theme: 'music' }]
 */
export function randMusic(options: RandThemedWordOptions & { output: 'detail' }): WordDetail[];
export function randMusic(options: RandThemedWordOptions = {}): string[] | WordDetail[] {
	return themedWord('music', options);
}

File: packages/dart/lib/src/word/rand_music.dart
import 'package:randino/src/types.dart';
import 'package:randino/src/word/rand_word.dart';

/// Music: instruments, forms and the words around them.
///
/// [randWord] with the theme already chosen; every other parameter is the same.
/// For the detail form, pass [WordTheme.music] to `randWordDetails` — Dart has
/// no overloads, and fourteen more functions for it would be fourteen too many.
///
/// ```dart
/// randMusic(language: WordLanguage.ko, count: 3); // [피아노, 거문고, 교향곡]
/// randMusic(language: WordLanguage.en, count: 3); // [Piano, Fiddle, Symphony]
/// ```
List<String> randMusic({
  WordLanguage? language,
  int count = 1,
  RandRealism realism = RandRealism.real,
  int? minLength,
  int? maxLength,
  String? startsWith,
  bool unique = false,
}) => randWord(
  language: language,
  theme: WordTheme.music,
  count: count,
  realism: realism,
  minLength: minLength,
  maxLength: maxLength,
  startsWith: startsWith,
  unique: unique,
);
